File: quarkus3-spring-compatibility/src/main/java/org/acme/service/FruitService.java
package org.acme.service;

import java.util.List;
import java.util.Optional;

import jakarta.transaction.Transactional;

import org.acme.dto.FruitDTO;
import org.acme.mapping.FruitMapper;
import org.acme.repository.FruitRepository;

import org.springframework.stereotype.Service;

@Service
public class FruitService {
  private final FruitRepository fruitRepository;

  public FruitService(FruitRepository fruitRepository) {
    this.fruitRepository = fruitRepository;
  }

  public List<FruitDTO> getAllFruits() {
    return this.fruitRepository.findAll().stream()
        .map(FruitMapper::map)
        .toList();
  }

  public Optional<FruitDTO> getFruitByName(String name) {
    return this.fruitRepository.findByName(name)
        .map(FruitMapper::map);
  }

  @Transactional
  public FruitDTO createFruit(FruitDTO fruitDTO) {
    var fruit = FruitMapper.map(fruitDTO);
    var savedFruit = this.fruitRepository.save(fruit);

    return FruitMapper.map(savedFruit);
  }
}

File: dotnet10/Dotnet10/Service/FruitService.cs
using Dotnet10.Dto;
using Dotnet10.Mapping;
using Dotnet10.Repository;

namespace Dotnet10.Service;

/// <summary>
/// Application service for Fruit operations.
/// Equivalent to @ApplicationScoped FruitService in the Quarkus project.
/// Transaction management is handled by the repository (SaveChangesAsync).
/// </summary>
public class FruitService(IFruitRepository fruitRepository)
{
    /// <summary>Returns all fruits as DTOs. Equivalent to getAllFruits() @Transactional(SUPPORTS).</summary>
    public async Task<List<FruitDto>> GetAllFruitsAsync() =>
        [.. (await fruitRepository.ListAllAsync())
            .Select(FruitMapper.Map)
            .OfType<FruitDto>()];

    /// <summary>
    /// Finds a fruit by name. Equivalent to getFruitByName() @Transactional(SUPPORTS).
    /// Returns null instead of Optional.empty() – idiomatic C# nullable reference type.
    /// </summary>
    public async Task<FruitDto?> GetFruitByNameAsync(string name)
    {
        var fruit = await fruitRepository.FindByNameAsync(name);
        return fruit is null ? null : FruitMapper.Map(fruit);
    }

    /// <summary>Creates and persists a new fruit. Equivalent to createFruit() @Transactional.</summary>
    public async Task<FruitDto> CreateFruitAsync(FruitDto fruitDto)
    {
        var fruit = FruitMapper.Map(fruitDto)!;
        await fruitRepository.PersistAsync(fruit);
        return FruitMapper.Map(fruit)!;
    }
}
